python/pyspark/errors/error_classes.py

```diff
@@ -59,6 +64,21 @@
       "Argument `<arg_name>` should be a bool, dict, float, int or str, got <arg_type>."
```

Comment:
should all these type have a different error classes? I was was thinking about adding these allowed types as a parameter, cc @MaxGekk FYI

Reply:
Yes. Since type-related error classes are increasing, it seems like it's time for refactoring.
Maybe there are two method in my mind for refactoring as below:

- We can create a main error class called such as `TYPE_MISMATCH` and convert all type-related errors to sub-error classes of `TYPE_MISMATCH`.
- Alternatively, as you suggested, we can create a `TYPE_MISMATCH` error class and add a parameter like `allowed_types` to manage all TYPE-related errors at once.

Which method do you think is more reasonable?

Reply:
Whatever matching to the current error classes in Scala would be nice.

Reply:
Okay, then let me go with option 1 even though the detail of the error definitions is a bit different.

Reply:
Can I do this in the separate ticket?

Reply:
Because applying the sub-error class can be a little complicated, so I want to proceed after finishing the rest of the PRs currently opened.